specs/CustomStoragePartition.md

```diff
@@ -0,0 +1,208 @@
+Custom Storage Partition
+===
+
+# Background
+For certain WebView host apps, there is a desire to have different contexts for
+different WebViews using the same profile so that when the same websites run in
+these WebViews, they will have different DOM storages and cookie jars. An example
+is different accounts within an application.
+Previosly, the application can use different profiles for different contexts, but
+that has 2 short comings:
+- WebViews from different profiles are not allowed to have opener/opened window relationship.
+- Using profiles means totally different storage for all files like http caches and less performance.
+
+To help managing different application contexts while using the same profile, we are
+introducing an API to set custom storage parition and an API to clear all data in
+a custom storage partition.
+
+When an application sets custom storage parition for a WebView, the sites and iframes
+running inside the WebView will act as if the site were running in a third party iframe
+inside a top level site uniquelly associated with the custom storage partition id and have
+separate [storage partition](https://developer.chrome.com/docs/privacy-sandbox/storage-partitioning/)
+and [cookie partition](https://developer.chrome.com/docs/privacy-sandbox/chips/).
+
+# Conceptual pages (How To)
+This feature depends on [storage partition](https://developer.chrome.com/docs/privacy-sandbox/storage-partitioning/)
+and [cookie partition](https://developer.chrome.com/docs/privacy-sandbox/chips/) browser
+features which are currently experimental. Before those features are enabled by default,
+the application must enable them by ensuring `--enable-features=ThirdPartyStoragePartitioning,PartitionedCookies`
+is set in [CoreWebView2EnvironmentOptions.AdditionalBrowserArguments](https://learn.microsoft.com/microsoft-edge/webview2/reference/winrt/microsoft_web_webview2_core/corewebview2environmentoptions#additionalbrowserarguments)
+used to create CoreWebView2Environment. If not set, no custom storage partition will be
+created and all data will be treated as unpartitioned and stored in the global default
+location for the profile.
+The custom storage partition APIs will remain experiemental when related browser features
+are still experimental.
+
+# Examples
+
+The example below illustrates how to set partition id on WebView and how to clear all
+data stored in the custom storage partition.
+
+## Win32 C++
+```cpp
+        wil::com_ptr<ICoreWebView2_18> m_webView;
+        
+        void OnWebViewCreated()
+        {
+           // ...
+           // other WebView setup like add event handlers, update settings.
+           
+           // Sets custom storage partition identified by the partitionId, which uniquely
+           // identifies an application context.
```

Comment:
Not sure what "uniquely identifies an application context." I thought it uniquely defined a custom storage partition.

Reply:
Yes, it uniquely identifies a custom storage partition, what I meant to state is that the unique partition is for an application context. Maybe could change it to something like: for a specific application context?

Reply:
What does application context mean?

Reply:
should I add "such as specific user account within the app", or simply remove the reference to application context?

Reply:
Please update: make it clear this is an app decision not required: "...which an app can use to uniquely identify..."